designs/sam_local_inlinecode.rst

```diff
@@ -0,0 +1,80 @@
+.. contents:: **Table of Contents**
+   :depth: 2
+   :local:
+
+``InlineCode`` debug
+=====================
+This is the design for a feature to provide local testing capabilities to functions defined via **InlineCode** key schema.
+
+What is the problem?
+--------------------
+When new functions are started an absolute path of the folder/file containing AWS Lambda code package need to be passed
+to docker deamon in order to bind-mount the path inside the container running the AWs Lambda runtime.
+
+As of today you can only test locally your AWS Lambda Functions if defined via ``CodeUri`` key and not ``InlineCode``.
+
+What will be changed?
+---------------------
+In this proposal, we will change the code in a way that, if the tested AWS Lambda function is codified with
+``InlineCode`` key, we will create a temporary local file based on the ``InlineCode`` value, passing the temp
+file to bind-mount inside the container and make it possible to locally test ``InlineCode`` authoer Functions.
+
+Success criteria for the change
+-------------------------------
+#. Support all programming languages supported by ``InlineCode``:
+   * Nodeds
+   * Python
+
+#. Local testing should just work for Functions authored with ``InlineCode``
+
+Out-of-Scope
+------------
+
+#. Anything not specifically related to ``InlineCode`` and sam local commands
+#. Provide support for Functions authored with ``InlineCode`` and leveraging ``!Sub`` pseudo function;
+   this can be object for future integration
+
+User Experience Walkthrough
+---------------------------
+
+
+Implementation
+==============
+
+Design
+------
+*Explain how this feature will be implemented. Highlight the components of your implementation, relationships*
+*between components, constraints, etc.*
+
+Design should be pretty simple. Write a tempfile and passing it to the Doker daemon as a bind-mount.
```

Comment:
You will run into the problem of Docker file sharing. If you create a temp folder using `tempfile.mkdtemp()`, chances are this creates within a folder that isn't shared in Docker. 

You have two options:
- Write the temp file somewhere within the project directory
- Or, use `docker cp` to copy the file into the container on startup. Since this file is likely to be very small, I am not concerned about the overhead added by this copy. The code already has a `copy` command that copies from within a container to host. You can follow this code to implement the reverse of it.

What do you think?

Reply:
Should we place these generated files into `./.aws-sam/build/<logical_id>` for inline code?

Reply:
Yes, I'm putting the file in ./.aws-sam/inline/<logical_id>/